controllers/storagecluster/cephcluster.go

```diff
@@ -528,17 +528,12 @@ func getNetworkSpec(sc ocsv1.StorageCluster) rookCephv1.NetworkSpec {
 	// respect both the old way and the new way for enabling HostNetwork
 	networkSpec.HostNetwork = networkSpec.HostNetwork || sc.Spec.HostNetwork
 
-	if networkSpec.Connections == nil {
-		networkSpec.Connections = &rookCephv1.ConnectionsSpec{}
-	}
-	networkSpec.Connections.RequireMsgr2 = true
-
-	// If it's an provider or external/consumer cluster don't require msgr2
-	// And don't allow encryption or compression to be enabled
-	if sc.Spec.AllowRemoteStorageConsumers || sc.Spec.ExternalStorage.Enable {
-		networkSpec.Connections.RequireMsgr2 = false
-		networkSpec.Connections.Encryption = nil
-		networkSpec.Connections.Compression = nil
+	// If it's not an external nor provider cluster then always require msgr2
+	if !sc.Spec.AllowRemoteStorageConsumers && !sc.Spec.ExternalStorage.Enable {
+		if networkSpec.Connections == nil {
+			networkSpec.Connections = &rookCephv1.ConnectionsSpec{}
+		}
+		networkSpec.Connections.RequireMsgr2 = true
```

Comment:
Do we need to override user input here can this be moved to line 535 to avoid overriding user input?

Reply:
I thought for normal case(if it's not external or provider), We want requiremsgr2 always on. But yeah I think not overriding user input is better. Changing it.

Reply:
Done PTAL